### PR TITLE
Add eslint

### LINKED ISSUE #174: Adding coding style checker like eslint?
Here is example.
```js
// toLint.js
1  console.log("Hello World");
2    var hi = "HHIHI";
3
4  function ji(){
5    return 123;
6  }
```

By using `eslint`, the example set style to `google`

```sh
$ eslint index.js

/tmp/index.js
  1:13  error  Strings must use singlequote              quotes
  2:3   error  Unexpected var, use let or const instead  no-var
  2:7   error  'hi' is assigned a value but never used   no-unused-vars
  2:12  error  Strings must use singlequote              quotes
  4:1   error  Missing JSDoc comment                     require-jsdoc
  4:10  error  'ji' is defined but never used            no-unused-vars
  4:14  error  Missing space before opening brace        space-before-blocks
```

- **Q1**: Do we need this?
- **Q2**: What style should be used?
  - airbnb
  - google
  - standard

Any suggestion?

### PR DESCRIPTION
**Pull request by drewfreyling**
fix current eslint issues:

-  around space-before-function
- paren and no-unused-vars

Fixes #174

### CODE REVIEW

**Comment by drewfreyling** [comment]
Once/if this is approved, we should add `npx eslint` to the CI pipeline.

File: webServer.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ throng({
   lifetime: Infinity
 }, start);
 
-function start() {
+function start () {
```
Comment (MattIPv4) [inline webServer.js]: I believe we currently only add a space on anonymous functions, is this something we can configure in eslint?

cc @PeterDaveHello

File: webServer.js [hunk 2]
```diff
@@ -270,7 +270,7 @@ function start() {
 
       if (assets.gennedFileNames === undefined) {
         var fileArray = [];
-        assets.files.map(function (fileName, index) {
+        assets.files.map(function (fileName) {
```
Comment (MattIPv4) [inline webServer.js]: Could this change be split into a separate commit?

One commit for the eslint file & changes to reflect the eslint enforcement, and a secondary commit for other tidying?
Or is this simply part of the eslint enforcement?

File: webServer.js [hunk 3]
```diff
@@ -348,7 +348,7 @@ function start() {
 
     var licenses = librarylicensesList(library);
     var assets = libraryAssetsList(library, version);
-    var stargazers_count = null, forks = null, subscribers_count = null;
+    var stargazers_count = null; var forks = null; var subscribers_count = null;
```
Comment (MattIPv4) [inline webServer.js]: I don't think should be a change that's occurring, is this something that eslint enforces?
I believe that using commas instead of a new var declaration for each one is cleaner.

cc @PeterDaveHello

File: webServer.js (hunk 2 above)
Comment (drewfreyling) [reply webServer.js]: I think its all part of eslint personally and we'd be splitting hairs to separate it out.

File: webServer.js (hunk 3 above)
Comment (drewfreyling) [reply webServer.js]: It was a hangover from when I was trialing eslint's standard guidelines. I've since reset the file and run a `--fix` and it doesn't want to change this line now.

File: webServer.js (hunk 1 above)
Comment (drewfreyling) [reply webServer.js]: Yup - i've modified it to `["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}]`

File: webServer.js (hunk 2 above)
Comment (MattIPv4) [reply webServer.js]: Ah cool, that's fine then :)

**Comment by drewfreyling** [comment]
> Lgtm.

Don't seem to have permissions anymore. I've requested access to the dev team for this.

**Comment by MattIPv4** [comment]
@drewfreyling I'm part of the dev team and can't merge either.
cc @PeterDaveHello ?

File: .eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    },
+    "parserOptions": {
+        "ecmaVersion": 2015
+    },
+    "rules": {
+        "space-before-function-paren": ["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}],
+         "no-console": "off"
+    },
+    "extends": "eslint:recommended"
+}
```
Comment (PeterDaveHello) [inline .eslintrc.json]: I think we should keep a final newline char here?

**Comment by PeterDaveHello** [comment]
I think @cdnjs/dev do have write access to this repo?

File: .eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    "env": {
+        "commonjs": true,
+        "node": true
+    },
+    "parserOptions": {
+        "ecmaVersion": 2015
+    },
+    "rules": {
+        "space-before-function-paren": ["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}],
+         "no-console": "off"
```
Comment (PeterDaveHello) [inline .eslintrc.json]: the indent is different here, I suggest to use 2 spaces, without any oneliner, you can use `jq` or `JSON.stringify()` to get a proper formatted one ;)

**Comment by MattIPv4** [comment]
cc #174

File: .eslintrc.json (hunk 5 above)
Comment (drewfreyling) [reply .eslintrc.json]: done.

File: .eslintrc.json (hunk 4 above)
Comment (drewfreyling) [reply .eslintrc.json]: done

**Comment by MattIPv4** [comment]
@drewfreyling Is it worth compacting your two commits down into one, dropping the merge commit and then rebasing against master, for a cleaner history?

**Comment by drewfreyling** [comment]
Why don't we just squash and merge?

**Comment by drewfreyling** [comment]
@MattIPv4 @PeterDaveHello cant' merge this - "The base branch restricts merging to authorized users.  Learn more about protected branches."

**Comment by MattIPv4** [comment]
@drewfreyling I can't merge either, only @PeterDaveHello can

File: package-lock.json [hunk 6]
```diff
@@ -0,0 +1,2029 @@
+{
```
Comment (PeterDaveHello) [inline package-lock.json]: This file should be removed here.

**Comment by MattIPv4** [comment]
My bad, we don't track the lock file, will use `npm i`.

File: .travis.yml [hunk 7]
```diff
@@ -17,6 +17,6 @@ jobs:
       env:
         - test: Editor Config check
```
Comment (MattIPv4) [inline .travis.yml]: @PeterDaveHello what is this env var used for?

Comment (PeterDaveHello) [reply .travis.yml]: Just help identify different test on the page

Comment (MattIPv4) [reply .travis.yml]: Ah, cool.

**Comment by PeterDaveHello** [comment]
I prefer to separate CI, eslint config and the fixes next time, anyway thanks for the help @MattIPv4
